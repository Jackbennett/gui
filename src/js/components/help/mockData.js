export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
