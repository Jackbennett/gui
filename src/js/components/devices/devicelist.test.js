import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import DeviceList from './devicelist';
import { undefineds } from '../../../../tests/mockData';

const mockStore = configureStore([thunk]);

describe('DeviceList Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      devices: {
        byId: {},
        filters: []
      },
      users: {
        onboarding: { complete: false }
      }
    });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DeviceList
            devices={[]}
            selectedRows={[]}
            columnHeaders={[{ name: 1 }, { name: 2 }, { name: 3 }, { name: 4 }]}
            columnWidth={100}
            pageLength={10}
            pageTotal={50}
          />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(JSON.stringify(tree)).toEqual(expect.not.stringMatching(undefineds));
  });
});
