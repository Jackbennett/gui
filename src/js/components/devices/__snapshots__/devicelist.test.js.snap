// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceList Component renders correctly 1`] = `
"<div>
  <div
    class=\\"deviceList \\"
  >
    <div
      class=\\"header deviceListRow columns-4 \\"
    >
      <div
        class=\\"columnHeader\\"
      />
      <div
        class=\\"columnHeader\\"
      />
      <div
        class=\\"columnHeader\\"
      />
      <div
        class=\\"columnHeader\\"
      />
      <div
        style=\\"width: 48px;\\"
      />
    </div>
    <div
      class=\\"body\\"
    />
    <div
      class=\\"flexbox margin-top\\"
    >
      <div
        class=\\"MuiTablePagination-root flexbox margin-top margin-top-none\\"
      >
        <div
          class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters\\"
        >
          <div
            class=\\"MuiTablePagination-spacer flexbox no-basis\\"
          />
          <p
            class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\"
            
          >
            Rows
          </p>
          <div
            class=\\"MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot\\"
          >
            <div
              aria-haspopup=\\"listbox\\"
              
              class=\\"MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input\\"
              
              role=\\"button\\"
              tabindex=\\"0\\"
            >
              20
            </div>
            <input
              aria-hidden=\\"true\\"
              class=\\"MuiSelect-nativeInput\\"
              tabindex=\\"-1\\"
              value=\\"20\\"
            />
            <svg
              aria-hidden=\\"true\\"
              class=\\"MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon\\"
              focusable=\\"false\\"
              viewBox=\\"0 0 24 24\\"
            >
              <path
                d=\\"M7 10l5 5 5-5z\\"
              />
            </svg>
          </div>
          <p
            class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit\\"
          />
          <div
            class=\\"flexbox\\"
          >
            <button
              class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
              disabled=\\"\\"
              tabindex=\\"-1\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiIconButton-label\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z\\"
                  />
                </svg>
              </span>
            </button>
            <button
              class=\\"MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled\\"
              disabled=\\"\\"
              tabindex=\\"-1\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiIconButton-label\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z\\"
                  />
                </svg>
              </span>
            </button>
            <div
              class=\\"flexbox\\"
              style=\\"align-items: baseline;\\"
            >
              <div
                class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-marginDense\\"
                style=\\"min-width: 30px; max-width: 3ch; margin-right: 10px;\\"
              >
                <div
                  class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense\\"
                >
                  <input
                    aria-invalid=\\"false\\"
                    class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense\\"
                    type=\\"text\\"
                    value=\\"1\\"
                  />
                </div>
              </div>
              / 3
            </div>
            <button
              class=\\"MuiButtonBase-root MuiIconButton-root\\"
              tabindex=\\"0\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiIconButton-label\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z\\"
                  />
                </svg>
              </span>
              <span
                class=\\"MuiTouchRipple-root\\"
              />
            </button>
            <button
              class=\\"MuiButtonBase-root MuiIconButton-root\\"
              tabindex=\\"0\\"
              type=\\"button\\"
            >
              <span
                class=\\"MuiIconButton-label\\"
              >
                <svg
                  aria-hidden=\\"true\\"
                  class=\\"MuiSvgIcon-root\\"
                  focusable=\\"false\\"
                  viewBox=\\"0 0 24 24\\"
                >
                  <path
                    d=\\"M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z\\"
                  />
                </svg>
              </span>
              <span
                class=\\"MuiTouchRipple-root\\"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
